slider ticket complete

jperez1478 opened the pull request:
Slider range value: 0.1601851569281684
Slider range value: 0.17314813401963974
Slider range value: 0.18981480068630643
Slider range value: 0.21574071248372395
Slider range value: 0.24166666666666667
Slider range value: 0.26296293470594617
Slider range value: 0.2740740458170573
Slider range value: 0.28703702290852867
Slider range value: 0.3111111111111111
Slider range value: 0.33703702290852866
Slider range value: 0.3740740458170573
Slider range value: 0.3990740458170573
Slider range value: 0.4166666666666667
Slider range value: 0.4361111111111111
Slider range value: 0.45555555555555555
Slider range value: 0.4814814673529731
Slider range value: 0.4990740458170573
Slider range value: 0.5083333333333333
Slider range value: 0.5120370229085286
Slider range value: 0.5148148006863065
Slider range value: 0.5185184902615018
Slider range value: 0.5222222222222223
Slider range value: 0.5277777777777778
Slider range value: 0.5407407124837239
Slider range value: 0.5518518235948351
Slider range value: 0.5777777777777777
Slider range value: 0.601851823594835
Slider range value: 0.6361111111111111
Slider range value: 0.6555555555555556
Slider range value: 0.6796296013726129
Slider range value: 0.7037036895751954
Slider range value: 0.7287036895751953
Slider range value: 0.7509259117974175
Slider range value: 0.775
Slider range value: 0.7981481340196398
Slider range value: 0.8222222222222222
Slider range value: 0.8462962680392795
Slider range value: 0.8527777495207611
Slider range value: 0.8555555272985389
Slider range value: 0.8574073791503907
Slider range value: 0.8583333050763167
Muted
Unmuted
Muted

jperez1478 committed:
slider ticket complete

jperez1478 committed:
fixed spacing

digimarktech commented on WhoopTriggerzDesign/CustomViews/CustomSliderModifier.swift:
```diff
@@ -0,0 +1,72 @@
+//
+import SwiftUI
+
+struct CustomSliderModifier: ViewModifier {
```
This shouldn't be a custom view modifier but instead a custom view. It would make sense to be a modifier if you were trying to create something like this:
```swift
Slider()
    .customSliderModifier()
```
Since this is not what we are trying to do, I would take all of the code from lines `16 to `64` and place that inside the body of a view. Something like this:
```swift
struct CustomSlider: View {
  var body: some View {
    // HStack code in here
  }
}
```

startdevelopfin replied:
Resolved in 1d0cfdf

digimarktech commented on WhoopTriggerzDesign/CustomViews/CustomSliderModifier.swift:
```diff
@@ -0,0 +1,72 @@
+struct CustomSliderModifier: ViewModifier {
+    //MARK: - Properties
+    @Binding var value : CGFloat
+    @State var isMuted: Bool = false
```
I think we might need to make the `isMuted` property a binding as well because we might need to communicate the value of this button being pressed to the parent view.
```suggestion
    @Binding var isMuted: Bool
```

startdevelopfin replied:
Resolved in 1d0cfdf

digimarktech commented on WhoopTriggerzDesign/CustomViews/CustomSliderModifier.swift:
```diff
@@ -0,0 +1,72 @@
+    //MARK: - Properties
+    @Binding var value : CGFloat
+    @State var isMuted: Bool = false
+    @State var speakerImage: String
```
Let's delete the `speakerImage` state variable because it's not needed. I'll suggest what we should do below
```suggestion
    
```

startdevelopfin replied:
Resolved in 1d0cfdf

digimarktech commented on WhoopTriggerzDesign/CustomViews/CustomSliderModifier.swift:
```diff
@@ -0,0 +1,72 @@
+            Button(action: {
+                // Speaker toggle button
+                isMuted.toggle()
+                speakerImage = isMuted ? "Mute" : "Volume"
```
We can delete this line as well
```suggestion
   
```

startdevelopfin replied:
Resolved in 1d0cfdf

digimarktech commented on WhoopTriggerzDesign/CustomViews/CustomSliderModifier.swift:
```diff
@@ -0,0 +1,72 @@
+                print(isMuted ? "Muted" : "Unmuted")
+                 
+            }) {
+                Image(speakerImage)
```
Instead of having the `speakerImage` state variable we had above, because `isMuted` is already a state variable, it will also cause the view to be redrawn whenever it's changed. On line `20`, we are already toggling the `isMuted` variable. Because of that, we can do the line I suggested below and it will work the same.
```suggestion
                Image(isMuted ? "Mute" : "Volume")
```

startdevelopfin replied:
Resolved in 1d0cfdf

digimarktech commented on WhoopTriggerzDesign/CustomViews/CustomSliderModifier.swift:
```diff
@@ -0,0 +1,72 @@
+                Image(speakerImage)
+                    .resizable()
+                    .frame(width: 30, height: 30)
+                    .padding()
```
We can remove this padding as it's not needed as is. If we need to add space between this image and our slider, since we only have 2 items in our `HStack`, we can set the spacing for the `HStack` or specify horizontal padding instead. We don't need padding all around
```suggestion
      
```

startdevelopfin replied:
Resolved in 1d0cfdf

digimarktech commented on WhoopTriggerzDesign/CustomViews/CustomSliderModifier.swift:
```diff
@@ -0,0 +1,72 @@
+            
+            ZStack(alignment: Alignment(horizontal: .leading, vertical: .center)) {
+                Capsule()
+                    .fill(Color.gray.opacity(0.2))
```
Based on the design, I think this color should be black based on this checklist item:
- Slider has black color that shows when dragged to the left

startdevelopfin replied:
Resolved in 1d0cfdf

digimarktech commented on WhoopTriggerzDesign/CustomViews/CustomSliderModifier.swift:
```diff
@@ -0,0 +1,72 @@
+                
+                ZStack(alignment: Alignment(horizontal: .trailing, vertical: .center)) {
+                    Capsule()
+                        .fill(Color.sliderColor.opacity(0.9))
```
Based on the design, I think this color should be a `LinearGradient`with that blueish color
- Slider has Linear Gradient that shows when dragged to the right

startdevelopfin replied:
Resolved in 1d0cfdf

digimarktech commented on WhoopTriggerzDesign/CustomViews/CustomSliderModifier.swift:
```diff
@@ -0,0 +1,72 @@
+                //divide and subtract lower bound upper bound of the range to get a value between 0.0 and 1.0
+                if dragValue.location.x <= UIScreen.main.bounds.width - 30 && dragValue.location.x >= 0{
+                    let rangeValue = Double((dragValue.location.x - 0) / (UIScreen.main.bounds.width - 30))
+                    //if it falls within the bounds print value
```
I think we might need to adjust the `30` value here because it's creating a bug. Notice that when you slide all the way to the right, the speaker button image seems to jump a little bit. I think that is because this `30` value needs to taking into account padding and some other items. I think it might need to be `55` or something like that. The view should not jump when the slider moves towards the end

startdevelopfin replied:
Resolved in 1d0cfdf

digimarktech commented on WhoopTriggerzDesign/Utilities/Colors+Ext.swift:
```diff
@@ -0,0 +1,13 @@
+import SwiftUI
+
+extension Color {
+    static let sliderColor = Color("blackColor")
```
I think we might need to adjust this value to better match the slider black color

startdevelopfin replied:
Resolved in 1d0cfdf

digimarktech reviewed (changes requested):
Awesome job getting this done @jperez1478!! Overall I like the implementation. I left a few comments that I think we should address before we merge this. I want to get as close to the design as possible because the owner of the app has a unique eye for certain details. I tried to leave as many examples as possible so please let me know if you have any questions

startdevelopfin committed:
Completes CustomSliderView

- Convert slider modifier to CustomSliderView
- Adds isMuted binding
- Updates slider track black color
- Adds blue Linear Gradient color to slider track

startdevelopfin committed:
Merge remote-tracking branch 'origin/main' into CustomSliderView

startdevelopfin committed:
Removing userstate
